kb/2023-12-27/best_practices_disk_performance.md

```diff
@@ -0,0 +1,60 @@
+slug: best_practices_for_optimizing_longhorn_disk_performance
+authors:
+  - name: David Ko
+    title: Senior Software Engineering Manager
+    url: https://github.com/innobead
+    image_url: https://github.com/innobead.png
+  - name: Jillian Maroket
+    title: Technical Writer
+    url: https://github.com/jillian-maroket/
+    image_url: https://github.com/jillian-maroket.png
+tags: [harvester, longhorn, best practices, disk performance]
+hide_table_of_contents: false
+---
+
+# Best Practices for Optimizing Longhorn Disk Performance
+
+The Longhorn documentation provides [best practice recommendations](https://longhorn.io/docs/1.5.3/best-practices/) for deploying Longhorn in production environments. Before configuring workloads, ensure that you have set up the following basic requirements for optimal disk performance.
+
+- SATA/NVMe SSDs or disk drives with similar performance
+- 10 Gbps network bandwidth between nodes
+- Dedicated Priority Classes for system-managed and user-deployed Longhorn components
+
+The following sections outline other recommendations for achieving optimal disk performance.
+
+## IO Performance
+
+- **Storage network**: Use a dedicated storage network to improve IO performance and stability.  
+
+- **Longhorn disk**: Use a dedicate for Longhorn storage instead of using the root disk.  
+
+- **Replica count**: Set the [default replica count](https://longhorn.io/docs/1.5.3/references/settings/#default-replica-count) to "2" to achieve data availability with better disk space usage or less impact to system performance. This practice is especially beneficial to data-intensive applications.  
+
+- **Storage tag**: Use [storage tags](https://longhorn.io/docs/1.5.3/volumes-and-nodes/storage-tags/) to define storage tiering for data-intensive applications. For example, only high-performance disks can be used for storing performance-sensitive data.  
+
+- **Data locality**: Use `best-effort` as the default [data locality](https://longhorn.io/docs/1.5.3/high-availability/data-locality/) of Longhorn Storage Classes.  
+
+  For applications that support data replication (for example, a distributed database), you can use the `strict-local` option to ensure that only one replica is created for each volume. This practice prevents the extra disk space usage and IO performance overhead associated with volume replication.  
+
+  For data-intensive applications, you can use pod scheduling functions such as node selector or taint toleration. These functions allow you to schedule the workload to a specific storage-tagged node together with one replica.  
+
+- **Revision counter**: You can disable the [revision counter](https://longhorn.io/docs/1.5.3/advanced-resources/deploy/revision_counter/) to improve IO performance, especially for write-intensive applications. When the revision counter is disabled, Longhorn does not track write operations for replicas and the Longhorn Engine does not check the `revision.counter` file after restarting.
```

Comment:
@innobead Do we want to recommend this for production environments?

Reply:
@innobead Can you comment on this?

Reply:
Let's remove this from this practice, as the focus here should be on actions that are 'recommended' be taken. 

However, this depends on a very strict requirement already mentioned in the 'important information' below, and it could be challenging to achieve in a prod setup. Additionally, it's not a recommendation but rather an optional method to enhance performance.

Reply:
I have removed this item.